src/screens/send/send.tsx

```diff
@@ -273,81 +331,92 @@ export class SendScreenComponent extends React.Component<
-                                style={styles.inputAddress}
-                                placeholderTextColor={theme.colors.textSecondary}
-                                placeholder={translate('Send.inputAddress')}
-                                autoCapitalize={'none'}
-                                autoCorrect={false}
-                                editable={!this.state.isValidAddress}
-                                selectionColor={theme.colors.accent}
-                                value={
-                                    this.state.isValidAddress
-                                        ? formatAddress(this.state.toAddress)
-                                        : this.state.toAddress
-                                }
-                                onChangeText={text => {
-                                    this.verifyAddress(text);
-                                }}
-                            />
-                            {this.renderRightAddressIcon()}
-                        </View>
-                        {this.state.labelErrorAddressDisplay && (
-                            <Text style={styles.displayError}>
-                                {translate('Send.recipientNotValid')}
-                            </Text>
-                        )}
-                        {this.state.labelWarningAddressDisplay && (
-                            <Text style={styles.receipientWarning}>
-                                {translate('Send.receipientWarning')}
-                            </Text>
-                        )}
-                        <TouchableOpacity
-                            testID="transfer-between-accounts"
-                            onPress={this.onTransferBetweenAccounts}
-                            style={[styles.buttonRightOptions]}
-                        >
-                            <Text style={styles.textTranferButton}>
-                                {this.state.showOwnAccounts
-                                    ? translate('App.labels.close')
-                                    : translate('Send.transferOwnAccounts')}
-                            </Text>
-                        </TouchableOpacity>
-
-                        {this.state.isValidAddress && this.renderBasicFields()}
-
-                        {this.state.showOwnAccounts && (
-                            <AccountList
-                                accounts={this.props.accounts}
-                                onAccountSelection={this.onAccountSelection}
-                            />
-                        )}
-
-                        <QrModalReader
-                            ref={ref => (this.qrCodeScanner = ref)}
-                            onQrCodeScanned={this.onQrCodeScanned}
+                    </TouchableOpacity>
+
+                    {this.state.isValidAddress &&
+                        !this.props.isContactAlreadySaved(this.state.toAddress) &&
```

Comment:
you don't really need a function here, you just need the contacts and you can do this check
```tsx
{this.state.isValidAddress &&
!this.props.contacts[`${this.props.account.blockchain}|${this.state.toAddress}`] && ... }
```

Also i would move the check in `renderAddAddressToBook`